src/kOS/Suffixed/PartModuleField/PartModuleFields.cs

```diff
@@ -235,8 +235,23 @@ public virtual BooleanValue HasField(StringValue fieldName)
         /// <returns>a BaseField - a KSP type that can be used to get the value, or its GUI name or its reflection info.</returns>
         protected BaseField GetField(string cookedGuiName)
         {
-            return partModule.Fields.Cast<BaseField>().
-                FirstOrDefault(field => string.Equals(GetFieldName(field), cookedGuiName, StringComparison.CurrentCultureIgnoreCase));
+            // Conceptually this should be a FirstOrDefault(), because there should only be one Field
+            // with the given GUI name.  But Issue #2666 forced kOS to change it to a list of hits
+            // because KSP started naming two fields with the same gui name, only one of which is visible
+            // at a time:
+            IEnumerable<BaseField> allMatches = partModule.Fields.Cast<BaseField>().
```

Comment:
You may want to call `.ToArray()` here to avoid a few unnecessary enumerations of the partModule fields. in the subsequent `Count()`'s

Reply:
Can you extrapolate on this?  Are you saying that partModule.Fields is of a type that requires a walk of the whole list to return a count?

Reply:
Ah I get it you were actually commenting on the line below this one.  I thought you were saying add a ToArray() to the line above, when actually you mean to the line below with the "where" clause.

Reply:
I think I understood what you asked for and put up an update to the PR - can you look at it?

I don't know if it's needed, though.  partModule.Fields is of type BaseFields, which already does implement its own Count() getter.  Without peeking at SQUAD's code illegally with a decompiler, I can't tell whether that Count() is just returning an already known integer or if it's running a walk, of the list to generate it.  I assume when you ask C# to do a Where(), the subset returned is the same type of collection as the original bigger collection, so I had assumed it was returning a BaseFields.

Reply:
It's honestly not really necessary just a small optimization. `.Count()` is a Linq extension method and will call the `IEnumerable<T>` methods `Current` and `MoveNext` to determine the count. [This SO post](https://stackoverflow.com/questions/4098186/lists-count-vs-count/4098586) explains the main gist. Linq typically has specialized versioned when the `IEnumerable<T>` is really a `ICollection<T>` or `IList<T>`. Since we call four additional methods on the `IEnumerable<T>` after the where save a little time.

Reply:
yeah the actual collection under the hood *does* have a simple integer count - the only question is whether or not Linq can detect that and use it, which I don't know.  At any rate I did change it.